syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/SyndesisExtensionProjectWizard.java

```diff
@@ -84,20 +85,22 @@ private NewSyndesisExtensionProjectMetaData getProjectMetaData() {
 		if (!locationPage.isInWorkspace()) {
 			metadata.setLocationPath(locationPage.getLocationPath());
 		}
-		metadata.setSyndesisExtensionConfig(getSyndesisExtension());
-		metadata.setTemplate(new BasicSyndesisExtensionXmlProjectTemplate());
+		metadata.setSyndesisExtensionConfig(extensionModel);
+		if (extensionDetailsPage.isCustomConnector()) {
```

Comment:
I think that for better separation of concerns, it would be better to have a SyndesisExtensionProjectWizardExtensionDetailsPage.getTemplate() method.
In this class we would have metadata.setTemplate(extensionDetailsPage.getTemplate()) only.
And when the page will evolve to provide more possibilities, the mdoificatiosn will be needed only in the wizard page and not both on wizard page class and in wizard class.
(can be done in a future iteration but seems easy to do it in this PR also)

Reply:
good suggestion